include/circt/Dialect/FSM/FSMOps.td

```diff
@@ -113,6 +114,24 @@ def TriggerOp : FSMOp<"trigger", []> {
   let hasVerifier = 1;
 }
 
+def ReferenceOp : FSMOp<"ref", []> {
+  let summary = "Reference some side effect through a symbol reference";
+  let description = [{
+    `fsm.ref` represents a reference to some symbol. The semantics of
+    a reference is implementation defined, and merely serves to convey
```

Comment:
I don't know if saying the semantics is implementation defined is really saying anything. Can you give a more specific semantics to what this op is, may or may not be used for, and why it would be used in the IR?